bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -135,7 +135,7 @@ public Client(@NonNull Context androidContext, @NonNull final Configuration conf
         storageManager = (StorageManager) appContext.getSystemService(Context.STORAGE_SERVICE);
 
         // Set up breadcrumbs
-        callbackState = configuration.callbackState.copy();
```

Comment:
I was expecting `ConfigImpl` to be used everywhere internally (rather than `Configuration`) - this is the advantage of retaining the Kotlin class: there should be no changes internally.

Reply:
This is not my understanding of the decorator pattern or the design that was sketched out, perhaps we can discuss what is required.

Reply:
As discussed IRL, we'll rename `ConfigImpl` to `ConfigInternal` and use `Configuration` to do _all_ validation but use `ConfigInternal` everywhere internally - this should avoid any change wider in library and maintain a more pure Kotlin class.

Reply:
Client should use the Internal class everywhere internally so that we get the benefits of nullability checks.

Reply:
I don't think this is necessary - we convert `Configuration` to an `ImmutableConfig` object whose properties cannot be mutated and are non-null where possible. We only need to access `Configuration` directly at the very start of the `Client` constructor and none of the properties used would gain any type-safety benefit from using `ConfigInternal` instead.